### DISCUSSION TITLE
create_directory_dialog just as  app.native.main_window.create_file_dialog?

### DISCUSSION THREAD

**Discussion by atticus-lv** [zauberzeug/nicegui#3100] [Ideas / Feature Requests]
hi there. Sometime I need to select the whole folder instead of a single file, so I can get what I need in the whole folder. maybe sometime to add a directory selector?

**Comment by frankvp11** [comment]
Perhaps you can make a tiny script using glob and just add the files individually? But I agree, a directory selector could be a nice addition, and probably wouldn't be too difficult to implement

**Comment by falkoschindler** [comment]
Interesting idea, @atticus-lv!

However, pywebview's way to select directories is using the `dialog_type` parameter:
```py
directory = await app.native.main_window.create_file_dialog(dialog_type=webview.FOLDER_DIALOG)
```
So I'm not sure if we should deviate from the original API by introducing new methods.

**Comment by atticus-lv** [comment]
wow, that is good. Maybe it can be add to the example / website for as a common usage.

**Comment by rustle123** [comment]
supporting folder upload seems a good idea, sometimes we'll need more files in a structured dir, hopefully this can be realized in the soon future, thanks a lot~
